app/src/main/java/com/android/unio/ui/association/AssociationProfile.kt

```diff
@@ -353,12 +465,86 @@ private fun AssociationProfileContent(
+ * should be displayed on the screen, such as the header, the description, the events...
+ *
+ * @param navigationAction [NavigationAction] : The navigation actions of the screen
+ * @param userViewModel [UserViewModel] : The user view model
+ * @param eventViewModel [EventViewModel] : The event view model
+ * @param associationViewModel [AssociationViewModel] : The association view model
+ */
+@Composable
+private fun AssociationProfileActionsContent(
+    navigationAction: NavigationAction,
+    userViewModel: UserViewModel,
+    eventViewModel: EventViewModel,
+    associationViewModel: AssociationViewModel,
+    searchViewModel: SearchViewModel
+) {
+  val context = LocalContext.current
+  val association by associationViewModel.selectedAssociation.collectAsState()
+  val user by userViewModel.user.collectAsState()
+
+  if (association == null || user == null) {
+    Log.e("AssociationProfileContent", "Association or user not found.")
+    return
+  }
+
+  var isFollowed by remember {
+    mutableStateOf(user!!.followedAssociations.contains(association!!.uid))
+  }
+  var enableButton by remember { mutableStateOf(true) }
+  val isConnected = NetworkUtils.checkInternetConnection(context)
+
+  val onFollow = {
+    if (isConnected) {
+      enableButton = false
+      associationViewModel.updateFollow(association!!, user!!, isFollowed) {
+        userViewModel.refreshUser()
+        enableButton = true
+      }
+      isFollowed = !isFollowed
+    } else {
+      ToastUtils.showToast(context, context.getString(R.string.no_internet_connection))
+    }
+  }
+
+  val onMemberClick = { member: User ->
+    userViewModel.setSomeoneElseUser(member)
+    navigationAction.navigateTo(Screen.SOMEONE_ELSE_PROFILE)
+  }
+
+  Column(
+      modifier =
+          Modifier.testTag(AssociationProfileActionsTestTags.SCREEN).fillMaxWidth().padding(24.dp),
+      verticalArrangement = Arrangement.spacedBy(16.dp)) {
+        AssociationActionsHeader(
+            association!!,
+            isFollowed,
+            enableButton,
+            onFollow,
+            onClickSaveButton = {
+              associationViewModel.selectAssociation(association!!.uid)
+              navigationAction.navigateTo(Screen.SAVE_ASSOCIATION)
+            })
+        AssociationActionsEvents(
+            navigationAction,
+            association!!,
+            userViewModel,
+            eventViewModel,
+            searchViewModel = searchViewModel)
+        AssociationActionsMembers(associationViewModel, user!!.uid, onMemberClick, searchViewModel)
+      }
+}
+
 /**
  * Component that displays the users that are in the association that can be contacted. It displays
  * the title of the section and then displays the different users in the association.
  *
  * @param members (List<User>) : The list of users in the association that can be contacted
  */
+@SuppressLint("StateFlowValueCalledInComposition")
```

Comment:
Consider removing this lint suppression and use the `by` operator when getting the user:
```kotlin
val user by associationViewModel.getUserFromMember(member).collectAsState()
```
Then, remove all `.value` calls.

Reply:
I don't understand how I can remove the .values calls but I did do the first two points ! Thanks !

Reply:
Oooo ok I get it ! Thanksss Now it's done ; )